### PR TITLE
Add CBMC proof-running GitHub Action

### PR DESCRIPTION
**Pull request by karkhaz**
This commit adds a GitHub Action that runs the CBMC proofs upon every push and pull request. This is intended to replace the current CBMC CI.

### CODE REVIEW

File: test/cbmc/proofs/run-cbmc-proofs.py [hunk 1]
```diff
@@ -334,20 +334,20 @@ async def configure_proof_dirs( # pylint: disable=too-many-arguments
...
-        "--phony-outputs", str(uuid.uuid4()),
-        "--pipeline-name", "print_tool_versions",
-        "--ci-stage", "report",
-        "--tags", "front-page-text",
-    ]
-    proc = subprocess.run(cmd)
-    if proc.returncode:
-        logging.critical("Could not add tool version printing job")
-        sys.exit(1)
+#def add_tool_version_job():
```
Comment (archigup) [inline test/cbmc/proofs/run-cbmc-proofs.py:337]: Why is this code commented out? Should it be removed?

Comment (karkhaz) [reply test/cbmc/proofs/run-cbmc-proofs.py:337]: Hi Archit, I'm planning on re-enabling this for all repositories (not all of them currently have it), but in a separate PR. Is that OK?

Comment (archigup) [reply test/cbmc/proofs/run-cbmc-proofs.py:337]: That is fine; in that case would be best for this PR to remove the commented out code, and can add it back in the other PR when ready

Comment (karkhaz) [reply test/cbmc/proofs/run-cbmc-proofs.py]: OK, done. Thank you

**Review by archigup — APPROVED** [review]

**Review by AniruddhaKanhere — APPROVED** [review]

**Pull request merged**